lib/geckoboard/datasets_client.rb

```diff
@@ -31,5 +31,13 @@ def put_data(dataset_id, data)
     def dataset_path(dataset_id)
       "/datasets/#{CGI.escape(dataset_id)}"
     end
+
+    def hashify_fields(fields)
+      return fields if fields.is_a? Hash
+
+      fields.inject({}) do |hash, field|
+        hash.merge(field.id => field.to_hash)
```

Comment:
What if `field` is not a `Field` descendant (i.e. does not respond to `#to_hash`)?